sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/ArrayBasedStack.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.state;
+
+import javax.annotation.Nullable;
+
+/**
+ * Array-based Stack.
+ *
+ * <p>Not thread-safe
+ */
+public class ArrayBasedStack<T> {
+  static final int DEFAULT_CAPACITY = 10;
+
+  // Using native array instead of ArrayList since I plan to add eviction
```

Comment:
Weird to have "I" in a comment without it being obvious who "I" refers to. Consider adding: `NOTE (asafm):` prefix